### ISSUE TITLE
Menlo integration

### ISSUE THREAD

**Issue by GINAGC** [k8-proxy/metadefender-menlo-integration#1]
Start analyzing how to  integrate with Menlo: https://github.com/OPSWAT/metadefender-menlo-integration

We will try to support 2 modes:
1.- The first one through Menlo 
2.- and the second one through OPSWAT (not a priority for now)

<img width="904" alt="1" src="https://user-images.githubusercontent.com/70196799/107364617-f70ae300-6adb-11eb-9552-533cb00ecbe0.png">

<img width="875" alt="2" src="https://user-images.githubusercontent.com/70196799/107364639-fc682d80-6adb-11eb-8f89-37668bdf0583.png">

**Comment by GINAGC** [comment]
Task paused and moved to NEW column.

**Comment by GINAGC** [comment]
Updates from developer:


**Created Meta-Defender Middleware:**
* Without Kubernetes,
* With Kubernetes (container env)
* The API endpoint is now available at: 3.14.72.244
* Changes: log variable in config.yml has to be created in advance before starting middleware server (will provide a script to do prerequisites steps once things are tested).
* Blocker: The value to be used in api url core  , from local it is easy to configure (windows) but from an ubuntu aws instance, still requires proper testing.

**Created Meta-Defender-Menlo integration:**
* created integration of Menlo security and MetaDefender Core using the above deployed middleware.
* This service will call the Sanitazation API, send tyhe file to metadefender api(above) for file analysis and send back the sanitized file.
* Again in the config.yml  core api url has to be specified (which is not clear), for now i have given it to be the one above  deployed middleware (3.14.72.244).
* Also i have created self signed certificate to the aws ubuntu server, as only https is allowed for any communication (required by Menlo).
* Configuring Menlo Integration:
* Blocker: Login as admin into https://admin.menlosecurity.com, we should provide company credentials to the admin UI, not sure if it is Menlo or any customer can create.

**Now for the testing part:**
It is said to navigate to: https://safe.menlosecurity.com
This also requires authentication which comes from the above mentioned step (Configuring Menlo Integration).
Also regarding Postman collection (https://github.com/OPSWAT/metadefender-menlo-integration/blob/master/api/Menlo%20S[…]urity%20-%20Sanitization%20Public%20API.postman_collection.json
) given, i will need some input since that also requires authentication which i think comes after steps mentioned :point_up_2: already.

**Comment by GINAGC** [comment]
* Previously i was getting timed out exception for the metadefender middleware api server requests (status: 403), now with some changes in the config.yml and running the middleware on windows (local without kubernetes), i am able to get specific http status code while running some commands with postman collection. (401,500, 304).
* This is the screenshot from the app logs which gets updated when i run a requests with postman collection(this is for "cloud" type) with the request to get already present file(like virustotal). - screenshot 1.
* The ideal response should be 404, but i get 403 in logs and 500 in postman, which is for unauthorized request, the thing is this does not require me to provide any UserID, need to check if they have enabled cloud access.
* Next is the app log screenshot with type as core which is available on the localhost as seen from the below image. The health-check returns 200 and the status returns 200 which says the middleware is running fine: screenshot 2 and 3.
* Now with the type set as "core" as in this config image, i get 500 error, with a proper json response of the error (unknown error) - screenshot 6 and 7.
* The thing is we should be able to see some success with the "check" request since they dont require userID, i feel there is some issue with the target server which is on port 8008, will check on this in detail in the morning.
